pgpubsub/tests/tests.py

```diff
@@ -87,7 +86,20 @@ def listener_func(
             deserialized,
         )
 
-    def _deserialize(self, callback, **kwargs):
-        channel = CustomPayloadChannel(callback)
-        serialized = _serialize(**kwargs)
-        return channel.deserialize(serialized)
+    def _deserialize(self, channel_cls, **kwargs):
+        serialized = channel_cls(**kwargs)._serialize()
+        return channel_cls.deserialize(serialized)
+
+
```

Comment:
@wesleykendall As mentioned on slack, I have my unit tests running on postgres in my local env. In the suite below, I'm attempting to test the listen-notify protocol. Whilst I can start listening to channels (line 98), invoking a notification to those channels (line 104) does not seem to result in any notifications being persisted to `self.pg_connection`. 

The only difference between the unit test set up below and how I've been testing it manually  so far is that when I tested it manually I started listening to the channels in a separate process (and did so using the `while True` loop in the `listen` function, which I obviously cannot do in a single threaded unit test).

Any ideas why this might not be working/ better ways to test this?

Reply:
This is the best I can get to work so far - basically we manually set the `notifies` attribute of the postgres connection. I think the test below essentially tests every aspect of the code, it just doesn't actually utilise the notify connecting the the listener
```
    def test_post_fetch_notify(self):
        author = Author.objects.create(name='Billy')
        today = datetime.date.today()
        post = Post.objects.create(
            author=author, content='first post', date=today)
        payload = notify(PostReads, model_id=post.pk, date=today)
        self.pg_connection.notifies = [
            psycopg2._psycopg.Notify(payload=payload, channel=PostReads.name(), pid='')]
        self.assertEqual(post_reads_per_date_cache[today], {})
        process_notifications(self.pg_connection)
        self.assertEqual(post_reads_per_date_cache[today], {post.pk: 1})

```

Reply:
Hmm actually this method of manually setting the self.pg_connection.notifies attribute will not work as well for testing trigger based notifications.